internal/oscal/types.go

```diff
@@ -0,0 +1,982 @@
+package oscal
+
+type OscalComponentDefinition struct {
+	Comment              string          `json:"$comment"`
+	ID                   string          `json:"$id"`
+	Schema               string          `json:"$schema"`
+	AdditionalProperties bool            `json:"additionalProperties"`
+	Definitions          Definitions_108 `json:"definitions"`
+	MaxProperties        int64           `json:"maxProperties"`
+	Properties           Properties_109  `json:"properties"`
+	Required             []string        `json:"required"`
+	Type                 string          `json:"type"`
+}
+
+type Items_45 struct {
```

Comment:
@brandtkeller would like to know what you think about the naming of the the structs

Seems a bit strange to me to have to reference numbers to use the types. The numbers reference the index number from the go slice that the struct was generated from

Any thoughts/preferences on the naming of the structs?

Reply:
Simply removing the numbers from the name isn't an option unfortunately, because there are duplicate names for multiple structs, and the go compiler (understandably) does not allow that. So there needs to be some sort of unique identifier for the structs

Reply:
Yeah - tracking the unique naming requirement. 

I'm going to look over the output currently - as all `Items_*` types except `Items_3` are almost the same - besides calling properties types that are each a little different.

Reply:
I may be well out of left-field here, but I believe that the current schema to type generation is generating to granularly? (if that makes sense).

example - when an object is defined- [like this](https://github.com/defenseunicorns/go-oscal/pull/1/files#diff-a205b615387acb399cf7db8522fddf334765de919015ee41b10daf466987e1aeR11) - The "properties" of said object are the the keys defined under "properties".

So in a component-definition object - we would not have:
```
component-definition:
  properties:
    UUID: 
```

rather we would have:
```
component-definition:
  UUID: 
```

But the JSON Schema is communicating that via a properties object. I believe most/all of these properties/items schema metadata structs would go away if we generated that with this in mind? Please feel free to say I explained it poorly or that I am wrong.

Reply:
Going to run some tests to see what it looks like to ingest and validate an example oscal component definition document with the generated types